gsr_booking/booking_logic.py

```diff
@@ -0,0 +1,255 @@
+            result_json["rooms"].append({"lid": lid, "room": room, "bookings": bookings_array})
+
+        # handle potential error
+        if error is not None:
+            if "not a valid".lower() in error.lower():
+                result_json["error"] = "Attempted to book for an invalid time slot"
+            elif "daily limit".lower() in error.lower():
+                result_json["error"] = "Group does not have sufficient booking credits"
+            else:
+                result_json["error"] = str(error)
+        return result_json
+
+    def get_used_booking_credit_for_user(self, lid, email):
+        # returns a user's used booking credit (in hours) for a specific building (lid)
+        RESERVATIONS_URL = "https://api.pennlabs.org/studyspaces/reservations"
+        if lid == "1":
+            return (False, 0)  # doesn't support huntsman yet
+        try:
+            r = requests.get(RESERVATIONS_URL + "?email=" + email)
+            if r.status_code == 200:
+                resp_data = r.json()
+                reservations = resp_data["reservations"]
+                used_credit_hours = 0
+                for reservation in reservations:
+                    from_date = datetime.datetime.strptime(
+                        reservation["fromDate"], self.DATE_FORMAT_STR
+                    )
+                    to_date = datetime.datetime.strptime(
+                        reservation["toDate"], self.DATE_FORMAT_STR
+                    )
+                    reservation_hours = (to_date - from_date).total_seconds() / 3600
+                    if str(reservation["lid"]) == lid and reservation_hours > 0.1:
+                        used_credit_hours += reservation_hours
+                return (True, used_credit_hours)
+        except requests.exceptions.RequestException as e:
+            print(e)
+        return (False, 0)
+
+    def split_booking(self, start, end, pennkey, booked):
+        # splits a booking into smaller bookings (of min_slot_hrs) for displaying to user
+        bookings = []
+        temp_start = start
+        temp_end = temp_start + datetime.timedelta(hours=self.MIN_SLOT_HRS)
+        while end - temp_start >= datetime.timedelta(hours=self.MIN_SLOT_HRS):
+            booking_obj = {
+                "start": temp_start.strftime(self.DATE_FORMAT_STR),
+                "end": temp_end.strftime(self.DATE_FORMAT_STR),
+                "booked": booked,
+            }
+            if pennkey is not None:
+                booking_obj["pennkey"] = pennkey
+            bookings.append(booking_obj)
+            temp_start = temp_end
+            temp_end += datetime.timedelta(hours=self.MIN_SLOT_HRS)
+        return bookings
+
+    def book_room_for_user(self, room, lid, start, end, email):
+        # tries to make a booking for an individual user, and returns success or not (and the error) in a tuple
+        if lid is "1":
+            return False  # does not support huntsman booking yet
```

Comment:
```suggestion
            return False, None  # does not support huntsman booking yet
```
So it works with the tuple and you don't get weird restructuring errors